packages/flutter/lib/src/material/text_selection.dart

```diff
@@ -66,33 +70,67 @@ class _TextSelectionToolbarState extends State<_TextSelectionToolbar> with Ticke
     );
   }
 
+  // Close the menu and reset layout calculations, as in when the menu has
+  // changed and saved values are no longer relevant. This should be called in
+  // setState or another context where a rebuild is happening.
+  void _reset() {
+    // Change _TextSelectionToolbarContainer's key when the menu changes in
+    // order to cause it to rebuild. This lets it recalculate its
+    // saved width for the new set of children, and it prevents AnimatedSize
+    // from animating the size change.
+    _containerKey = UniqueKey();
+    // If the menu items change, make sure the overflow menu is closed. This
+    // prevents an empty overflow menu.
+    _overflowOpen = false;
+  }
+
+  // Inform the widget that the value of clipboardStatus has changed.
```

Comment:
nit: place this comment inside the empty setState callback to explain why the callback is empty.